packages/common/src/discussions/edit.ts

```diff
@@ -98,6 +117,26 @@ export async function updateDiscussion(
   // now map back into a discussion and return that
   let updatedDiscussion = mapper.storeToEntity(updatedModel, discussion);
   updatedDiscussion = computeProps(model, updatedDiscussion, requestOptions);
+  if (updatedDiscussion.discussionSettings) {
```

Comment:
Similarly, we need to update `requestOptions` arg type to `IHubRequestOptions` here and spread it onto the object passed to `createSetting` and `updateSetting` for auth to work for the settings API requests. It's also not enough to check if `discussionSettings` exists to make the determination if we need to call `updateSetting` or `createSetting` since we provide default values.

```jsx
export async function updateDiscussion(
  discussion: IHubDiscussion,
  requestOptions: IHubRequestOptions
): Promise<IHubDiscussion> {

  // ...
  
  // update the backing item
  const updatedModel = await updateModel(modelToUpdate, requestOptions as IUserRequestOptions);
  // now map back into a discussion and return that
  let updatedDiscussion = mapper.storeToEntity(updatedModel, discussion);
  updatedDiscussion = computeProps(model, updatedDiscussion, requestOptions);
  if (updatedDiscussion.entitySettingsId) {
    updateSetting({
      id: updatedDiscussion.entitySettingsId,
      data: {
        settings: {
          discussions: updatedDiscussion.discussionSettings,
        },
      },
      ...requestOptions,
    });
  } else {
    const defaultSettings = getDefaultEntitySettings('discussion');
    const entitySettings = await createSetting({
      data: {
        id: updatedDiscussion.id,
        type: defaultSettings.type,
        settings: {
          ...defaultSettings.settings,
          discussions: {
            ...defaultSettings.settings.discussions,
            ...updatedDiscussion.discussionSettings,
          },
        },
      },
      ...requestOptions,
    });
    updatedDiscussion.entitySettingsId = entitySettings.id;
    updatedDiscussion.discussionSettings = entitySettings.settings.discussions;
  }
```